src/json_printer.sql

```diff
@@ -55,40 +70,65 @@ package body "JSON_PRINTER" as
       return str;
     end if;
   end llcheck;  
-
-  function escapeString(str varchar2) return varchar2 as
-    sb varchar2(32767) := '';
-    buf varchar2(40);
-    num number;
+  
+  -- escapes a single character. 
+  function escapeChar(ch char) return varchar2 deterministic is
```

Comment:
All I have asked is that you refactor a portion of your patch to match a preferred style. Unless you can provide some measurements that show the function is an order of magnitude faster with a return statement in each `WHEN` clause, that's all that needs to be done for the patch to be accepted.